{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d0d02b3b-eb5a-4bbe-b014-1d44f4b01af9",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import scipy.special\n",
    "import glob\n",
    "from matplotlib import colormaps\n",
    "import matplotlib.colors as mcolors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3a0093c3-d9d0-47bc-8221-e1dc5618d8e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get the colors and genus in descending order as previously defined in script: M2_DefineColors\n",
    "goodColors = np.load('/home/isabel/Documents/postDoc_Amsterdam/B_BioInfo/1_MyScripts/GitHub/HostMicrobiomeEvol/Metagenomics/2_Analysis/M2_colors_ColorsRGB.npy')\n",
    "goodColorsHEX = np.load('/home/isabel/Documents/postDoc_Amsterdam/B_BioInfo/1_MyScripts/GitHub/HostMicrobiomeEvol/Metagenomics/2_Analysis/M2_colors_ColorsHEX.npy')\n",
    "genus_descending = np.load('/home/isabel/Documents/postDoc_Amsterdam/B_BioInfo/1_MyScripts/GitHub/HostMicrobiomeEvol/Metagenomics/2_Analysis/M2_colors_genusDescending.npy')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "15e62790-3c33-4cbf-83e2-c7b71e5c1452",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name_short</th>\n",
       "      <th>institution</th>\n",
       "      <th>treatment</th>\n",
       "      <th>replicate</th>\n",
       "      <th>week</th>\n",
       "      <th>Abyssalbus</th>\n",
       "      <th>Abyssibius</th>\n",
       "      <th>Acetobacter</th>\n",
       "      <th>Acetobacterium</th>\n",
       "      <th>Acetobacteroides</th>\n",
       "      <th>...</th>\n",
       "      <th>Yoonia</th>\n",
       "      <th>Youhaiella</th>\n",
       "      <th>Zhihengliuella</th>\n",
       "      <th>Zhongshania</th>\n",
       "      <th>Zobellella</th>\n",
       "      <th>Zobellia</th>\n",
       "      <th>Zophobihabitans</th>\n",
       "      <th>Zunongwangia</th>\n",
       "      <th>Zymobacter</th>\n",
       "      <th>Zymomonas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Gent_bb_1_1</td>\n",
       "      <td>Gent</td>\n",
       "      <td>bb</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00004</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Gent_bb_1_2</td>\n",
       "      <td>Gent</td>\n",
       "      <td>bb</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Gent_bb_1_3</td>\n",
       "      <td>Gent</td>\n",
       "      <td>bb</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gent_bb_1_4</td>\n",
       "      <td>Gent</td>\n",
       "      <td>bb</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00005</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gent_bb_15_1</td>\n",
       "      <td>Gent</td>\n",
       "      <td>bb</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>WUR_bb_15_1</td>\n",
       "      <td>WUR</td>\n",
       "      <td>bb</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>WUR_bb_15_2</td>\n",
       "      <td>WUR</td>\n",
       "      <td>bb</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>WUR_bb_15_3</td>\n",
       "      <td>WUR</td>\n",
       "      <td>bb</td>\n",
       "      <td>3</td>\n",
       "      <td>15</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>WUR_bleach_15_1</td>\n",
       "      <td>WUR</td>\n",
       "      <td>bleach</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>WUR_bleach_15_2</td>\n",
       "      <td>WUR</td>\n",
       "      <td>bleach</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>93 rows Ã— 1254 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         name_short institution treatment  replicate  week  Abyssalbus  \\\n",
       "0       Gent_bb_1_1        Gent        bb          1     1         0.0   \n",
       "1       Gent_bb_1_2        Gent        bb          2     1         0.0   \n",
       "2       Gent_bb_1_3        Gent        bb          3     1         0.0   \n",
       "3       Gent_bb_1_4        Gent        bb          4     1         0.0   \n",
       "4      Gent_bb_15_1        Gent        bb          1    15         0.0   \n",
       "..              ...         ...       ...        ...   ...         ...   \n",
       "88      WUR_bb_15_1         WUR        bb          1    15         0.0   \n",
       "89      WUR_bb_15_2         WUR        bb          2    15         0.0   \n",
       "90      WUR_bb_15_3         WUR        bb          3    15         0.0   \n",
       "91  WUR_bleach_15_1         WUR    bleach          1    15         0.0   \n",
       "92  WUR_bleach_15_2         WUR    bleach          2    15         0.0   \n",
       "\n",
       "    Abyssibius  Acetobacter  Acetobacterium  Acetobacteroides  ...  Yoonia  \\\n",
       "0          0.0          0.0             0.0               0.0  ...     0.0   \n",
       "1          0.0          0.0             0.0               0.0  ...     0.0   \n",
       "2          0.0          0.0             0.0               0.0  ...     0.0   \n",
       "3          0.0          0.0             0.0               0.0  ...     0.0   \n",
       "4          0.0          0.0             0.0               0.0  ...     0.0   \n",
       "..         ...          ...             ...               ...  ...     ...   \n",
       "88         0.0          0.0             0.0               0.0  ...     0.0   \n",
       "89         0.0          0.0             0.0               0.0  ...     0.0   \n",
       "90         0.0          0.0             0.0               0.0  ...     0.0   \n",
       "91         0.0          0.0             0.0               0.0  ...     0.0   \n",
       "92         0.0          0.0             0.0               0.0  ...     0.0   \n",
       "\n",
       "    Youhaiella  Zhihengliuella  Zhongshania  Zobellella  Zobellia  \\\n",
       "0          0.0         0.00004          0.0         0.0       0.0   \n",
       "1          0.0         0.00003          0.0         0.0       0.0   \n",
       "2          0.0         0.00003          0.0         0.0       0.0   \n",
       "3          0.0         0.00005          0.0         0.0       0.0   \n",
       "4          0.0         0.00000          0.0         0.0       0.0   \n",
       "..         ...             ...          ...         ...       ...   \n",
       "88         0.0         0.00002          0.0         0.0       0.0   \n",
       "89         0.0         0.00001          0.0         0.0       0.0   \n",
       "90         0.0         0.00002          0.0         0.0       0.0   \n",
       "91         0.0         0.00003          0.0         0.0       0.0   \n",
       "92         0.0         0.00002          0.0         0.0       0.0   \n",
       "\n",
       "    Zophobihabitans  Zunongwangia  Zymobacter  Zymomonas  \n",
       "0               0.0           0.0         0.0        0.0  \n",
       "1               0.0           0.0         0.0        0.0  \n",
       "2               0.0           0.0         0.0        0.0  \n",
       "3               0.0           0.0         0.0        0.0  \n",
       "4               0.0           0.0         0.0        0.0  \n",
       "..              ...           ...         ...        ...  \n",
       "88              0.0           0.0         0.0        0.0  \n",
       "89              0.0           0.0         0.0        0.0  \n",
       "90              0.0           0.0         0.0        0.0  \n",
       "91              0.0           0.0         0.0        0.0  \n",
       "92              0.0           0.0         0.0        0.0  \n",
       "\n",
       "[93 rows x 1254 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in data to panda\n",
    "\n",
    "InPath=\"/home/isabel/Documents/postDoc_Amsterdam/1_EvolWormJourney/1_Genomics/2_EvolGenomics/2_Metagenomics/1_taxonomy/\"\n",
    "#file=\"motus_Merge_genus_032024_0.01minFrac_FromPY\"\n",
    "file=\"KraBracken_DBstandard_042024_0minFrac_genus_FromPY\"\n",
    "#file=\"KraBracken_DBplusPf_042024_0.01minFrac_genus_FromPY\"\n",
    "\n",
    "df=pd.read_csv(InPath+file+\".csv\",delimiter=\",\")\n",
    "\n",
    "threshold = 0.01\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34811b8b-5f0b-4642-bbe4-ae3417acec0b",
   "metadata": {},
   "source": [
    "# Threshold:\n",
    "Here, for every plot we ask which species is at least in 1 sample at the threshold and then we plot these species for all the samples. This means that we do show sometimes species that are there less than 0.01. Why do we do this? Otherwise it would look as if the sample doesn't have the species and maybe that would be misleasding?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "e75e4d67-7be4-48e5-8461-e45b46f9b546",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.9555800000000001"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df_plot_Here = df_sort\n",
    "##filter=('bb')\n",
    "df_plot = df[df[\"week\"]==15]\n",
    "df_plot_drop=df_plot.drop(['name_short',\"institution\",\"treatment\",\"replicate\",\"week\"],axis=1)\n",
    "\n",
    "#keep only the columns that are larger than or equal to 0.01\n",
    "df_plot_temp = df_plot_drop.loc[:, (df_plot_drop >= 0.001).any(axis=0)]\n",
    "\n",
    "## now get the species and colors that are relevant for this plot!\n",
    "\n",
    "## Sort the df according to the list genus_descending AND\n",
    "## -- make sure that the colors for the species that don't exist are deleted:\n",
    "species_here = list(df_plot_temp.columns)\n",
    "\n",
    "species_idx = []\n",
    "for i in range(len(genus_descending)):\n",
    "    if genus_descending[i] in species_here:\n",
    "        species_idx.append(i)\n",
    "aa = df_plot_temp[df_plot_Here[\"name_short\"]==\"Gent_bb_15_2\"]\n",
    "aa_clean = aa##.drop(['name_short',\"institution\",\"treatment\",\"replicate\",\"week\"],axis=1)\n",
    "aaa = np.array(aa_clean.values)\n",
    "aaa[0].sum()\n",
    "#df_plot_temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b4e760cc-24d6-4a26-84d2-4060525fe212",
   "metadata": {},
   "outputs": [],
   "source": [
    "##PLOT\n",
    "\n",
    "# get the names and the values and then sort both\n",
    "\n",
    "df_plot_Here = df_plot[['name_short'] + list(genus_descending[species_idx])]\n",
    "finalColors_Here = goodColors[species_idx]\n",
    "\n",
    "# Generate some sequential data\n",
    "sampleNames=df_plot_Here['name_short']\n",
    "\n",
    "\n",
    "#fig, ax1 = plt.subplots(figsize=(20,5))\n",
    "# create stacked bar chart for monthly temperatures\n",
    "ax1 = df_plot.plot.bar(x='name_short', stacked=True, figsize=(20,10), width=0.9)#, color=finalColors_Here, figsize=(20,10), width=0.9)\n",
    "ax1.legend([],[], frameon=False)\n",
    "\n",
    "\n",
    "ax1.legend(loc='upper center', bbox_to_anchor=(0.5, 1.1),\n",
    "          ncol=6, fancybox=True, shadow=True)\n",
    "\n",
    "\n",
    "#ax.set_ylim(0,500)\n",
    "#ax1.set_xticks([0,1,2,3,4])\n",
    "ax1.set_xticklabels(sampleNames,rotation=45)\n",
    "#ax1.set_xlim([-1,5])\n",
    "ax1.set_ylim([0,1.07])\n",
    "plt.savefig('/home/isabel/Documents/postDoc_Amsterdam/1_EvolWormJourney/1_Genomics/2_EvolGenomics/2_Metagenomics/1_taxonomy/plots/' + '_'.join(filter) + '_legend.svg',\n",
    "            dpi=200)\n",
    "df_plot_Here\n",
    "\n",
    "aa = df_plot_Here[df_plot_Here[\"name_short\"]==\"Gent_bb_15_2\"]\n",
    "aa_clean = aa.drop(['name_short'],axis=1)\n",
    "aaa = np.array(aa_clean.values)\n",
    "aaa[0].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "d6e1abd4-0343-43c5-a762-2c53d21f856b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[0.19921875, 0.1328125 , 0.53125   , 1.        ],\n",
       "       [0.06640625, 0.46484375, 0.19921875, 1.        ],\n",
       "       [0.59765625, 0.59765625, 0.19921875, 1.        ],\n",
       "       [0.796875  , 0.3984375 , 0.46484375, 1.        ],\n",
       "       [0.53125   , 0.1328125 , 0.33203125, 1.        ],\n",
       "       [0.6640625 , 0.265625  , 0.59765625, 1.        ],\n",
       "       [0.86328125, 0.86328125, 0.86328125, 1.        ],\n",
       "       [0.53125   , 0.796875  , 0.9296875 , 1.        ],\n",
       "       [0.265625  , 0.6640625 , 0.59765625, 1.        ],\n",
       "       [0.9296875 , 0.796875  , 0.3984375 , 1.        ],\n",
       "       [0.9296875 , 0.19921875, 0.46484375, 1.        ],\n",
       "       [0.        , 0.        , 0.        , 1.        ],\n",
       "       [0.99609375, 0.6640625 , 0.73046875, 1.        ],\n",
       "       [0.59765625, 0.46484375, 0.        , 1.        ],\n",
       "       [0.59765625, 0.265625  , 0.33203125, 1.        ],\n",
       "       [0.9296875 , 0.46484375, 0.19921875, 1.        ],\n",
       "       [0.46484375, 0.69921875, 0.86328125, 1.        ],\n",
       "       [0.796875  , 0.19921875, 0.06640625, 1.        ],\n",
       "       [0.265625  , 0.73046875, 0.59765625, 1.        ],\n",
       "       [0.8671875 , 0.19140625, 0.38671875, 1.        ],\n",
       "       [0.59765625, 0.86328125, 0.99609375, 1.        ],\n",
       "       [0.6640625 , 0.6640625 , 0.        , 1.        ],\n",
       "       [0.9296875 , 0.86328125, 0.53125   , 1.        ],\n",
       "       [0.9296875 , 0.53125   , 0.3984375 , 1.        ],\n",
       "       [0.99609375, 0.6640625 , 0.73046875, 1.        ],\n",
       "       [0.86328125, 0.86328125, 0.86328125, 1.        ],\n",
       "       [0.73046875, 0.796875  , 0.9296875 , 1.        ],\n",
       "       [0.796875  , 0.9296875 , 0.99609375, 1.        ],\n",
       "       [0.796875  , 0.86328125, 0.6640625 , 1.        ],\n",
       "       [0.9296875 , 0.9296875 , 0.73046875, 1.        ],\n",
       "       [0.99609375, 0.796875  , 0.796875  , 1.        ],\n",
       "       [0.1328125 , 0.1328125 , 0.33203125, 1.        ],\n",
       "       [0.1328125 , 0.33203125, 0.33203125, 1.        ],\n",
       "       [0.1328125 , 0.33203125, 0.1328125 , 1.        ]])"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "goodColors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48403a06-a019-4d09-8912-c9d1064b4b07",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5258a40e-6ed1-4f08-9068-3e53f0a56543",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
